File: Banda_Beat_Android/src/com/irontec/bandabeat/db/Playlist.java
package com.irontec.bandabeat.db;

public class Playlist {
	
	private int idPlaylist;
	private String name;
	private int songCount;
	private String token;
	private String songToken;
	
	
	public int getIdPlaylist() {
		return idPlaylist;
	}
	public void setIdPlaylist(int idPlaylist) {
		this.idPlaylist = idPlaylist;
	}
	public String getName() {
		return name;
	}
	public void setName(String name) {
		this.name = name;
	}
	public int getSongCount() {
		return songCount;
	}
	public void setSongCount(int songCount) {
		this.songCount = songCount;
	}
	public String getToken() {
		return token;
	}
	public void setToken(String token) {
		this.token = token;
	}
	public String getSongToken() {
		return songToken;
	}
	public void setSongToken(String songToken) {
		this.songToken = songToken;
	}
	
	
}

File: Banda_Beat_iOS/Banda Beat/Playlist.m
//
//  Playlist.m
//  Banden Lehia
//
//  Created by Iker Mendilibar on 30/10/12.
//  Copyright (c) 2012 Irontec S.L. All rights reserved.
//

#import "Playlist.h"
#import "Track.h"
#import "NSString+IMAddition.h"


@implementation Playlist

@dynamic downloaded;
@dynamic idPlaylist;
@dynamic name;
@dynamic token;
@dynamic songCount;
@dynamic songToken;
@dynamic tracks;
@dynamic dirty;


-(void)setDataWithdictionary:(NSDictionary*)data
{
    NSString *name= [data objectForKey:@"name"];
    
    if ([NSString isEmptyString:name]) {
        name = @"Sin nombre";
    }
    
    [self setName:name];
    [self setIdPlaylist:[data objectForKey:@"idPlaylist"]];
    [self setToken:[data objectForKey:@"token"]];
    [self setSongToken:[data objectForKey:@"songToken"]];
    [self setSongCount:[data objectForKey:@"songCount"]];
    [self setDownloaded:[NSNumber numberWithBool:NO]];
    [self setDirty:[NSNumber numberWithBool:NO]];
}

@end
